server/embed/config_logging.go

```diff
@@ -165,35 +165,38 @@ func (cfg *Config) setupLogging() error {
 			return err
 		}
 
-		logTLSHandshakeFailure := func(conn *tls.Conn, err error) {
-			state := conn.ConnectionState()
-			remoteAddr := conn.RemoteAddr().String()
-			serverName := state.ServerName
-			if len(state.PeerCertificates) > 0 {
-				cert := state.PeerCertificates[0]
-				ips := make([]string, len(cert.IPAddresses))
-				for i := range cert.IPAddresses {
-					ips[i] = cert.IPAddresses[i].String()
+		logTLSHandshakeFailureFunc := func(msg string) func(conn *tls.Conn, err error) {
+			return func(conn *tls.Conn, err error) {
+				state := conn.ConnectionState()
+				remoteAddr := conn.RemoteAddr().String()
+				serverName := state.ServerName
+				if len(state.PeerCertificates) > 0 {
+					cert := state.PeerCertificates[0]
```

Comment:
Could we ever be in a state where `PeerCertificates` other than `[0]` should be inspected?

Reply:
I didn't change this. I just set different warning message on client and peer endpoints in this PR.

Going back to your question, `state.PeerCertificates[0]` is the leaf certificate which we should check/verify against, so I think it's expected.

Reply:
Yes not a change introduced by this pr, it was just a question that crossed my mind as I was reviewing overall.  Thanks for clarifying it.